backend/python/app/utilities/csv_utils.py

```diff
@@ -120,7 +120,10 @@ def generate_csv_from_list(dict_list, **kwargs):
     :return: csv string
     :rtype: str
     """
-    field_names = dict_list[0].keys()
+    field_names = {
+        key: None
+        for d in dict_list for key in d.keys()
+    }.keys()
```

Comment:
First item doesn't necessarily have all the keys that need to be included (for example, `file_name` in entity is an optional field). Will sync the fix with `main` in the upcoming bug fix PR.